src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -32,19 +32,29 @@ public void TestEnvironmentProperty()
 
             IDictionary<string, string> environment = psi.Environment;
 
-            Assert.NotEqual(environment.Count, 0);
+            Assert.NotEqual(0, environment.Count);
 
-            int CountItems = environment.Count;
+            int countItems = environment.Count;
 
             environment.Add("NewKey", "NewValue");
             environment.Add("NewKey2", "NewValue2");
 
-            Assert.Equal(CountItems + 2, environment.Count);
+            Assert.Equal(countItems + 2, environment.Count);
             environment.Remove("NewKey");
-            Assert.Equal(CountItems + 1, environment.Count);
+            Assert.Equal(countItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
```

Comment:
Just to make sure we're talking about the same thing:
It looks like when you do `processStartInfo.Environment.Add` and the key already exists, in desktop it overwrites the existing key/value whereas currently in core it throws.  That's a big change, introducing an exception where there wasn't one before, and in a scenario that's quite easy to hit.  Why is the core behavior easier/better?  Porting that back to desktop would almost certainly break code and would I expect be a non-starter.